src/System.Memory/src/System/SequencePosition.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+
+namespace System
+{
+    /// <summary>
+    /// Represents position in non-contiguous set of memory.
+    /// Properties of this type should not be interpreted by anything but the but the
+    /// type that created it.
+    /// </summary>
+    public readonly struct SequencePosition : IEquatable<SequencePosition>
+    {
+        /// <summary>
+        /// Creates new <see cref="SequencePosition"/>
+        /// </summary>
+        public SequencePosition(object segment, int index)
+        {
+            Segment = segment;
```

Comment:
Should we have _segment/_index as private fields of the struct?

Reply:
I made them properties because of the comment above.